src/utils.jl

```diff
@@ -56,11 +58,51 @@ end
 
 # family of projection functions
 
-immutable Proj{field}; end
+immutable Proj{F}
+    f::F
+end
+
+(p::Proj)(x::Tup) = p.f(x)
+function (p::Proj{Val{f}}){f}(x::Tup)
+    getfield(x, f)
+end
 
-(::Proj{field}){field}(x) = getfield(x, field)
+"""
+    @pick(fields...)
+
+Returns a callable object `f` such that `f(x::Tuple)` returns a `Tuple` with only
+elements of index specified by `fields`, `f(x::NamedTuple)` return a `Tuple` if
+`fields` are integers, or a `NamedTuple` if `fields` are symbols with only the specified
+fields in the output.
+
+The callable is specialized to work efficiently on `Columns` by calling it once
+on `.columns` field to get the equivalent result.
+
+Calling `map` on an `IndexedTable` with a `@pick` callable will run the callable on
+the data columns.
+
+# Examples
+    c = Columns(x=[1], y=[2.0])
+    @pick(2)(c) == Columns([2.0])
+    @pick(y)(c) == Columns(y=[2.0])
+    @pick(2,1)(c) == Columns([2.0], [1])
+    @pick(y,x)(c) == Columns(y=[2.0], x=[1])
+
+    t = IndexedTable([1], c)
+    map(@pick(y, x), t) == IndexedTables([1], Columns(y=[2.0], x=[1]))
+"""
+macro pick(ex...)
+    tup = if all([isa(x, Symbol) for x in ex])
+        # Named tuple
+        T = NamedTuples.create_tuple([x for x in ex])
+        :(NamedTuples.$T($([:(getfield(x, $(Expr(:quote, f)))) for f in ex]...)))
```

Comment:
>  it will be sent to other processes

I'm trying to make this change to NamedTuples but it doesn't look like we can send new types on `release-0.5` branch of Julia:

```
> jl -p 1

julia> struct Foo
           x
       end

julia> remotecall_fetch(identity, 2, Foo("x"))
ERROR: On worker 2:
UndefVarError: Foo not defined
deserialize_datatype at ./serialize.jl:861
handle_deserialize at ./serialize.jl:601
deserialize at ./serialize.jl:571
ntuple at ./tuple.jl:108
handle_deserialize at ./serialize.jl:592
deserialize_msg at ./distributed/messages.jl:100
deserialize_msg at ./distributed/messages.jl:110
message_handler_loop at ./distributed/process_messages.jl:161
process_tcp_streams at ./distributed/process_messages.jl:118
```